app/app/core/security.py

```diff
@@ -8,20 +8,46 @@
 
 pwd_context = CryptContext(schemes=["bcrypt"], deprecated="auto")
 
+ALGORITHM = "HS512"
 
-ALGORITHM = "HS256"
 
-
-def create_access_token(subject: Union[str, Any], expires_delta: timedelta = None) -> str:
+def create_token(subject: Union[str, Any], expires_delta: timedelta = None, type_: str = 'access') -> str:
     if expires_delta:
         expire = datetime.utcnow() + expires_delta
     else:
-        expire = datetime.utcnow() + timedelta(minutes=settings.ACCESS_TOKEN_EXPIRE_MINUTES)
-    to_encode = {"exp": expire, "sub": str(subject)}
+        expire = datetime.utcnow() + get_timedelta_for_type(type_)
```

Comment:
can we put UTC datetime of expiry here instead of delta? that would be easier to handle at the frontend.

I can use the [jwt-decode](https://www.npmjs.com/package/jwt-decode) library in the react web client to decode the token, currently getting an int as the value for `exp` which is the timedelta from token creation to expiry. That makes it hard to track on persistent sessions, since the original time of token creation is not known to client.

Reply:
I'd rather pass the expiry time separately in the response, and not bother about decoding the JWT in the frontend.